beacon-chain/blockchain/process_block.go

```diff
@@ -366,12 +366,24 @@ func (s *Service) databaseDACheck(ctx context.Context, b consensusblocks.ROBlock
 
 func (s *Service) updateEpochBoundaryCaches(ctx context.Context, st state.BeaconState) error {
 	e := coreTime.CurrentEpoch(st)
+	// The proposer indices cache takes the target root for the previous
+	// epoch as key
 	if err := helpers.UpdateCommitteeCache(ctx, st, e); err != nil {
 		return errors.Wrap(err, "could not update committee cache")
 	}
 	if err := helpers.UpdateProposerIndicesInCache(ctx, st, e); err != nil {
 		return errors.Wrap(err, "could not update proposer index cache")
 	}
+	// The latest block header is from the previous epoch
+	r := [32]byte(st.LatestBlockHeader().BodyRoot)
```

Comment:
This looks wrong. Why are we using the `BodyRoot` all of a sudden? Shouldn't it be `HashTreeRoot`?